File: CSharp/K11.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

namespace CSharp
{
    class K11 : IExecutable
    {
        private const int MaxPeople = 5;

        public override void Question01()
        {
            try
            {
                List<int> ages = new List<int>();

                for (int i = 0; i < MaxPeople; i++)
                {
                    Console.Write((i + 1) + "人目の年齢を入力して下さい：");
                    ages.Add(int.Parse(Console.ReadLine()));
                }

                Console.WriteLine($"{ages.Count}人の平均年齢は{ages.Average()}です。");
            }
            catch (Exception e)
            {
                Console.Error.WriteLine(e.Message);
            }
        }

        public override void Question02()
        {
            try
            {
                Console.Write("1つ目の文字列を入力して下さい。＞");
                string str1 = Console.ReadLine();
                Console.Write("2つ目の文字列を入力して下さい。＞");
                string str2 = Console.ReadLine();
                str1 += str2;
                Console.WriteLine($"{str1}の文字数は{str1.Length}です。");
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
            }
        }

        public override void Question03()
        {
            try
            {
                char chr;
                Console.Write("文字を入力して下さい。＞");
                chr = char.Parse(Console.ReadLine());
                bool applied = false;

                if (char.IsDigit(chr))
                {
                    Console.WriteLine("この文字は数字です。");
                    applied = true;
                }

                if (char.IsUpper(chr))
                {
                    Console.WriteLine("この文字は英大文字です。");
                    applied = true;
                }

                if (char.IsLower(chr))
                {
                    Console.WriteLine("この文字は英小文字です。");
                    applied = true;
                }

                Regex rx = new Regex("^[A-Fa-f0-9]{6}$");

                if (rx.IsMatch(chr.ToString()))
                {
                    Console.WriteLine("この文字は16進数として使える文字です。");
                    applied = true;
                }

                if (char.IsPunctuation(chr))
                {
                    Console.WriteLine("この文字は区切り文字です。");
                    applied = true;
                }

                if (!applied)
                {
                    Console.WriteLine("この文字はよくわからない文字です。");
                }

            }
            catch (Exception e)
            {
                Console.WriteLine(e);
            }
        }

        public override void Question04()
        {
            string alphabets = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

            string inputString = "I am a STUDENT. I am good at GAME PROGRAMMING.";

            int count = 0;
            int _count = 0;
            Console.WriteLine($"文字列：{inputString}\n");
            Console.WriteLine("文字頻度");
            alphabets.ToList().ForEach((x) => {
                inputString.ToList().ForEach((y) =>
                {
                    if (x.ToString() == y.ToString().ToUpper())
                    {
                        count++;
                    }
                });
                Console.Write($"{x}: {count}\t");
                count = 0;
                _count++;

                if (_count % 4 == 0)
                {
                    Console.WriteLine();
                }
            });
        }
    }
}

File: Kotlin/src/K11.kt
import java.util.*
import java.util.regex.Pattern

class K11 : IExecutable() {
    override fun question_01() {
        val ages = Vector<Int>()
        val max_people = 5

        val reader = Scanner(System.`in`)
        for (i in 0 until max_people) {
            print("${i + 1}人目の年齢を入力して下さい：")
            ages.add(reader.nextInt())
        }

        val _average = ages.stream().mapToDouble { x -> x.toDouble() }.average()
        var average = 0.0
        if (_average.isPresent) {
            average = _average.asDouble
        }

        println("${ages.size}人の平均年齢は${average}です。")
    }

    override fun question_02() {
        print("1つ目の文字列を入力して下さい。＞")
        val reader = Scanner(System.`in`)
        var str1 = reader.nextLine()
        print("2つ目の文字列を入力して下さい。＞")
        val str2 = reader.nextLine()
        str1 += str2
        println("${str1}の文字数は${str1.length}です。")
    }

    override fun question_03() {
        val chr: Char
        print("文字を入力して下さい。＞")
        val reader = Scanner(System.`in`)
        chr = reader.nextLine()[0]
        var applied = false

        if (chr.isDigit()) {
            println("この文字は数字です。")
            applied = true
        }
        if (chr.isUpperCase()) {
            println("この文字は英大文字です。")
            applied = true
        }
        if (chr.isLowerCase()) {
            println("この文字は英小文字です。")
            applied = true
        }
        var pattern = Pattern.compile("^[A-Fa-f0-9]$")
        var matcher = pattern.matcher(chr.toString())
        if (matcher.matches()) {
            println("この文字は16進数として使える文字です。")
            applied = true
        }
        pattern = Pattern.compile("^(.|,|!|\\?|:|;|\"|'|\\|-|\\(|\\))$")
        matcher = pattern.matcher(chr.toString())
        if (matcher.matches()) {
            println("この文字は区切り文字です。")
            applied = true
        }

        if (!applied) {
            println("この文字はよくわからない文字です。")
        }
    }

    override fun question_04() {
        val alphabets = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        val inputString = "I am a STUDENT. I am good at GAME PROGRAMMING."
        var count = 0
        var _count = 0
        println("文字列：${inputString}\n")
        println("文字頻度")
        for (s in alphabets) {
            for (_s in inputString) {
                if (s == _s.toUpperCase())
                    count++
            }
            print("${s}: ${count}\t")
            count = 0
            _count++
            if (_count % 4 == 0)
                println()
        }
    }

}